doc/tutorials/run_all_tests_with_docker.md

```diff
@@ -0,0 +1,133 @@
+# Introduction
+
+Running all the tests like the build server requires to have multiple dependencies installed. To overcome this problem, instead of trying to install all the necessary dependencies on your own, an appropriate Docker image can be used. This way you can easily and quickly run all the tests.
+
+## Who is this guide for?
+
+For anyone who wants to run all the tests, like it is done by the build server.
+
+This is a step-by-step guide. Just follow the steps and you are good to go!
+
+## Prerequisites
+
+- Docker for Linux containers has to be preinstalled. Please refer to https://docs.docker.com/install/ if you haven't installed it yet. Your host OS can be either Linux or Windows of course.
+- Basic knowledge of Docker (not mandatory)
+
+## What to begin with?
+
+### 1. Pick a Docker image and pull it
+
+Pick one of the available Docker images of Elektra. If you do not know the difference, just pick this one --> "build-elektra-debian-stretch".
+Unfortunately, it will take some time to download it, since it is pretty big, but you can be sure you'll have all the needed dependencies.
+You can choose a light-weight alpine image which won't take long to download, however it is not recommended. This image does not contain all necessary dependencies.
+
+If you want to view all the available images, execute this command:
+
+```sh
+docker run --rm anoxis/registry-cli -r https://hub-public.libelektra.org
+```
+
+You will see something like this:
+
+```sh
```

Comment:
Please remove the incorrect language identifier (`sh`). The code below is not valid [(Bourne) shell code](https://en.m.wikipedia.org/wiki/Bourne_shell).

Reply:
Should I just leave the triple braces? Like this?
```
output example
```

Reply:
Yes. Initially I wanted to add a [“suggested change”](https://help.github.com/en/articles/incorporating-feedback-in-your-pull-request#applying-a-suggested-change) for that update too. However, I do not think this is possible, since the [characters ` ``` `  are also used to end a suggestion](https://help.github.com/en/articles/commenting-on-a-pull-request#adding-line-comments-to-a-pull-request).